cluster/v1beta1/helpers.go

```diff
@@ -317,16 +292,56 @@ func (pdct *PlacementDecisionClustersTracker) fulfillGroupKeys(groupKeys []Group
 	return fulfilledGroupKeys
 }
 
-func (pdct *PlacementDecisionClustersTracker) getOrderedGroupKeysBesides(orderedGroupKeyToExclude []GroupKey) []GroupKey {
-	orderedGroupKey := []GroupKey{}
+func (pdct *PlacementDecisionClustersTracker) getGroupKeysBesides(groupKeyToExclude []GroupKey) []GroupKey {
+	groupKey := []GroupKey{}
 	for i := 0; i < len(pdct.clusterGroupsIndexToName); i++ {
-		groupKey := GroupKey{GroupName: pdct.clusterGroupsIndexToName[int32(i)], GroupIndex: int32(i)}
-		if !containsGroupKey(orderedGroupKeyToExclude, groupKey) {
-			orderedGroupKey = append(orderedGroupKey, groupKey)
+		gKey := GroupKey{GroupName: pdct.clusterGroupsIndexToName[int32(i)], GroupIndex: int32(i)}
+		if !containsGroupKey(groupKeyToExclude, gKey) {
+			groupKey = append(groupKey, gKey)
 		}
 	}
 
-	return orderedGroupKey
+	return groupKey
+}
+
+// ClusterGroupsMap is a custom type representing a map of group keys to sets of cluster names.
+type ClusterGroupsMap map[GroupKey]sets.Set[string]
+
+// GetOrderedGroupKeys returns an ordered slice of GroupKeys, sorted by group index.
+func (g ClusterGroupsMap) GetOrderedGroupKeys() []GroupKey {
+	groupKeys := []GroupKey{}
+	for groupKey := range g {
+		groupKeys = append(groupKeys, groupKey)
+	}
+
+	// sort by group index index
+	sort.Slice(groupKeys, func(i, j int) bool {
+		return groupKeys[i].GroupIndex < groupKeys[j].GroupIndex
+	})
+
+	return groupKeys
+}
+
+// GetClusterSets returns a set containing all clusters from all group sets.
+func (g ClusterGroupsMap) GetClusterSets() sets.Set[string] {
```

Comment:
ClusterSets here is confusing, GetClusters maybe

Reply:
Fixed.